src/screens/qr/components/Toolbar.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useCallback} from 'react';
+import {StyleSheet} from 'react-native';
+import {Box, Button} from 'components';
+import {useNavigation} from '@react-navigation/native';
+import {useI18n} from 'locale';
+import {BackButton} from '../views/BackButton';
+
+export interface ToolbarProps {
+  onIconClicked(): void;
+  navText?: string;
+  showBackButton?: boolean;
+  showCloseButton?: boolean;
+  isWhite?: boolean;
+}
+
+export const Toolbar = ({navText, onIconClicked, showBackButton, showCloseButton, isWhite}: ToolbarProps) => {
+  return (
+    <Box flexDirection="row" alignItems="center" minHeight={56}>
+      {showBackButton ? <ToolbarWhite isWhite={isWhite} /> : null}
+      {showCloseButton ? (
+        <Box style={styles.right}>
+          <Button testID="toolbarCloseButton" text={navText} variant="text" onPress={onIconClicked} />
+        </Box>
+      ) : null}
+    </Box>
+  );
+};
+
+export const ToolbarWhite = ({isWhite}: {isWhite?: boolean}) => {
+  const navigation = useNavigation();
+  const i18n = useI18n();
+  return isWhite ? (
+    <BackButton
+      textStyles={styles.backText}
+      iconName="icon-chevron-back-white"
+      text={i18n.translate(`QRCode.Reader.Back`)}
+      onPress={() => {
+        navigation.navigate('Home');
```

Comment:
This is the back button, so the function should be `navigation.goBack()`. Also it should be wrapped in a `useCallback` so that the function doesn't have to be re-assigned to the variable whenever the component re-renders.